frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/confirm.tsx

```diff
@@ -42,24 +42,17 @@ export async function loader({ request }: LoaderFunctionArgs) {
   await raoidcService.handleSessionValidation(request);
 
   const userId = await userService.getUserId();
-
-  if (!userId) {
-    instrumentationService.countHttpStatus('phone-number.confirm', 302);
-    return redirectWithLocale(request, '/');
-  }
-
   const userInfo = await userService.getUserInfo(userId);
   if (!userInfo) {
-    instrumentationService.countHttpStatus('phone-number.confirm', 302);
-    return redirectWithLocale(request, '/');
+    instrumentationService.countHttpStatus('home-address.edit', 404);
+    throw new Response(null, { status: 404 });
   }
 
   const session = await sessionService.getSession(request);
   if (!session.has('newPhoneNumber')) {
-    instrumentationService.countHttpStatus('phone-number.confirm', 302);
-    return redirectWithLocale(request, '/');
+    instrumentationService.countHttpStatus('home-address.edit', 404);
+    throw new Response(null, { status: 404 });
```

Comment:
Does a 404 make sense here? I think if a `newPhoneNumber` is not found in session, it makes might make sense to bounce the user back to the phone number edit page (since this is where the `newPhoneNumber` session variable is set).

Also, the metric name should be `phone-number.confirm` (with the correct status -- make sure to change it to 302 if that's what we decide to do).

Reply:
Requesting @sebastien-comeau or @nicholas-ly for comment.

Reply:
In this case I think it makes sense to redirect to the edit screen.